base/common/CMakeLists.txt

```diff
@@ -66,7 +66,17 @@ if(WITH_TEST)
             pki-certsrv-jar
     )
 
-    # TODO: create CMake function to find all JUnit test classes
+    execute_process(
+        COMMAND bash "-c"
+        "grep -ilR @Test ${PROJECT_SOURCE_DIR} \
+        | cut -d':' -f1 \
+        | awk -F '/src/test/java/' '{ print $2 }' \
+        | sed 's/.java/;/g' \
+        | sed 's!/!.!g' \
+        | tr -d '\n'"
+        OUTPUT_VARIABLE DISCOVERED_TESTS
+    )
+    
```

Comment:
Noice! :)

If we can assume that all Java files in the test folder are test files maybe it can be simplified further.

Just FYI, we have this code to find all Java sources in a directory:
https://github.com/dogtagpki/pki/blob/master/cmake/Modules/Java.cmake#L76-L81
https://github.com/dogtagpki/pki/blob/master/cmake/Modules/JavaFileList.cmake
but no need to use this if yours work.

Reply:
Noice, didn't know about this source finding function! The majority of the new function is massaging the discovered tests package name into a format that is acceptable to `cmake`. Finding the tests is an easy `grep` for the annotation, so I don't think we gain much by abstracting that part out into another function.